tests/unit/authentication_test.py

```diff
@@ -2,27 +2,63 @@
 
 from werkzeug import exceptions
 from mock import MagicMock
+from http_server_mock import HttpServerMock
+from cryptography.hazmat.primitives.asymmetric import rsa
+import json
+import jwt
+from jwt.utils import to_base64url_uint
+import time
 
 from bond_app.authentication import Authentication, AuthenticationConfig, UserInfo
 from bond_app.sam_api import SamApi, SamKeys
 from tests.unit.fake_cache_api import FakeCacheApi
-import json
 
+ALGORITHM = "RS256"
+PUBLIC_KEY_ID = "sample-key-id"
 
 class TestRequestState:
     def __init__(self, auth_header):
         self.headers = {}
         if auth_header is not None:
             self.headers['Authorization'] = auth_header
 
+class MockOidcServer:
```

Comment:
Creates a mock JWKS endpoint to be used for encoding and decoding valid JWTs.

Used https://dev.to/filipemir/mocking-auth0-tokens-in-python-and-beyond-5dki as a guide.